Common/Product/SharedProject/ProjectNode.cs

```diff
@@ -634,7 +634,6 @@ internal enum EventTriggering {
                     projectHome = CommonUtils.GetAbsoluteDirectoryPath(
                         this.ProjectFolder,
                         this.GetProjectProperty(CommonConstants.ProjectHome, resetCache: false));
-                    projectHome = CommonUtils.TrimEndSeparator(projectHome);
```

Comment:
This change will likely have much wider effects than the other two. Does it need to be changed or is the IsSameDirectory change sufficient?